components/authentication-framework/org.wso2.carbon.identity.application.authentication.framework/src/main/java/org/wso2/carbon/identity/application/authentication/framework/cache/SessionContextCache.java

```diff
@@ -75,6 +80,25 @@ public void addToCache(SessionContextCacheKey key, SessionContextCacheEntry entr
         entry.setAccessedTime();
         super.addToCache(key, entry, resolveLoginTenantDomain(loginTenantDomain));
         Object authUser = entry.getContext().getProperty(FrameworkConstants.AUTHENTICATED_USER);
+        if (isSessionDataStorageOptimizationEnabled) {
```

Comment:
Isn't this condition always true?

Reply:
Changed to get config from IdentityUtil

Reply:
Fixed with [`87d8c01`](https://github.com/wso2/carbon-identity-framework/pull/4446/commits/87d8c01eaa49ef513f71a55b77154408818a31b3)